### PR TITLE
Add compatibility with base 4.9

### PR DESCRIPTION
**Pull request by Wysaard**
These commits will allow one to build every package except for `gloss-examples` with base 4.9. The examples still work in 4.8 so no functionality is broken, only added.

The reason the examples can't work is because as of yet there seems to be no version of GLFW-b compatible with the new base. However, a version is not forced so without changes to the code it should start working as soon as such an update is released.

In summary:
On 4.8 everything should work the same. On 4.9 you can now use everything except the examples, where previously the build failed.

### CODE REVIEW

**Comment by Wysaard** [comment]
Similar changes were made by @benl23x5 ; this pull is no longer needed.
